includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -702,7 +702,7 @@ private function is_product_supported() {
 		}
 
 		// Composite products are not supported on the product page.
-		if ( class_exists( 'WC_Composite_Products' ) && function_exists( 'is_composite_product' ) && is_composite_product() ) {
+		if ( class_exists( 'WC_Composite_Products' ) && $product->is_type( 'composite' ) ) {
```

Comment:
Thanks for making this change!

Reply:
I hope you checked backward compatibility. I haven't found exact version of WC when this method appeared, but based on [this](https://woocommerce.github.io/code-reference/files/woocommerce-includes-abstracts-abstract-wc-product.html#source-view.1490) I assume it's there since 3.0 so it's safe to use.

Reply:
Interesting point @vbelolapotkov ! 

Without a doc-block is tough to say when this method was introduced. However, this method is used in other places in this file as well. See: https://github.com/Automattic/woocommerce-payments/blob/16bc82b81914554cfef55f2a328677e28a2679a3/includes/class-wc-payments-payment-request-button-handler.php#L710

Doing a search in the Woo core repo, I found that unit tests for `is_type` were added in v2.3: https://github.com/woocommerce/woocommerce/blob/3611d4643791bad87a0d3e6e73e031bb80447417/plugins/woocommerce/tests/legacy/unit-tests/product/product-simple.php#L87. So, it is safe to assume that this method was added either before v2.3 or at v2.3.

Reply:
Ok, good. Feels safe then 😄